pkg/apiserver/registry/inventory/virtualmachine/rest.go

```diff
@@ -164,9 +170,40 @@ func (r *REST) ConvertToTable(_ context.Context, obj runtime.Object, _ runtime.O
 
 func (r *REST) Watch(ctx context.Context, options *internalversion.ListOptions) (watch.Interface, error) {
 	key, label, field := store.GetSelectors(options)
+	contextNamespace, _ := request.NamespaceFrom(ctx)
+	if key != "" {
+		if contextNamespace == "" {
+			return nil, errors.NewBadRequest("cannot watch for a resources in all namespaces")
+		} else {
+			key = contextNamespace + "/" + key
+		}
+	} else if contextNamespace != "" {
+		// Namespace specified and field selector doesn't have namespace.
+		if _, ok := field.RequiresExactMatch("metadata.namespace"); !ok {
```

Comment:
Why do we need explicit handling for metadata.namespace?

The keyAndSpanSelectFuncVm is anyways going to match the object against with fieldselectors and labelselectors.

Reply:
If -n is specified, I set field selector only when field selector doesn't have metadata.namespace set. If I don't have the check then -n ns1 --field-selector metadata.namespace=ns2, ns2 will be overwritten. In this case I am giving ns2 priority over ns1. Although it's a invalid case.